lib/command/src/Obelisk/Command/Project.hs

```diff
@@ -193,6 +214,17 @@ walkToImplDir projectRoot myUid = do
 isWritableOnlyBy :: FileStatus -> UserID -> Bool
 isWritableOnlyBy s uid = fileOwner s == uid && fileMode s .&. 0o22 == 0
 
+isOwnedBy :: FileStatus -> UserID -> Bool
+isOwnedBy s uid = fileOwner s == uid
+
+-- | Check to see if a directory respect the umask, but check explicitly that
+-- it's not world writable in any case.
+respectsUmask :: FileStatus -> FileMode -> Bool
+respectsUmask s umask = isNotWorldWritable && umaskRespected
+  where
+    isNotWorldWritable = fileMode s .&. 0o02 == 0
+    umaskRespected = fileMode s .&. umask == 0
```

Comment:
It seems it would be theoretically possible for a file to *not* be world writable but for the umask to make files world writable. In this case we should be fine with the file since it's not *actually* world-writable.